python/ray/actor.py

```diff
@@ -287,24 +291,28 @@ def temporary_actor_method(*xs):
 
 
 def register_actor_signatures(worker, driver_id, class_name,
-                              actor_method_names):
+                              actor_method_names,
+                              actor_method_num_return_vals):
     """Register an actor's method signatures in the worker.
 
     Args:
         worker: The worker to register the signatures on.
         driver_id: The ID of the driver that this actor is associated with.
         actor_id: The ID of the actor.
         actor_method_names: The names of the methods to register.
+        actor_method_num_return_vals: A list of the number of return values for
+            each of the actor's methods.
     """
-    for actor_method_name in actor_method_names:
+
+    for actor_method_name, num_return_vals in zip(
```

Comment:
Maybe add an assertion that `len(actor_method_names) == len(actor_method_num_return_vals)`

Reply:
Fixed